compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1890,29 +1890,29 @@ object Types {
     }
 
     /** The signature of the last known denotation, or if there is none, the
-     *  signature of the symbol
+     *  signature of the symbol. Signatures are always computed before erasure, since
+     *  some symbols change their signature at erasure.
      */
-    protected def computeSignature(implicit ctx: Context): Signature = {
+    protected def computeSignature(implicit ctx: Context): Signature =
       val lastd = lastDenotation
-      if (lastd != null) lastd.signature
+      if lastd != null && lastd.validFor.firstPhaseId <= ctx.erasurePhase.id then lastd.signature
+      else if ctx.erasedTypes then computeSignature(using ctx.withPhase(ctx.erasurePhase))
       else symbol.asSeenFrom(prefix).signature
-    }
 
     /** The signature of the current denotation if it is known without forcing.
```

Comment:
This documentation is now incorrect since this is the signature of the pre-erasure denotation. However this method is exclusively used to overloading resolution, so it should never be called after erasure I think, so maybe we can get away with leaving the documentation as is and adding an `assert(!ctx.erasedTypes)`